### PR TITLE
API-24660: Automate OAS Diff Deployment

### PR DESCRIPTION
**Pull request by jeremyhwu**
* Moved oasdiff files from /dev/oasdiff to /src/ui/oasdiff
* Upload oasdiff.js/oasdiff.css files to S3 on server startup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -261,54 +285,65 @@ A sample kong service and OAS are provided for local end-to-end testing. The `do
...
 
 ### Viewing OAS Iteration Differences
-When DocServer detects a meaningful change to an OAS a new iteration is stored in the database.  Along with this change a static 'oasdiff' HTML file will be generated to make viewing the series of changes overtime easier.
 
-For local DEV environment these static files can be viewed with the following URL pattern: 
+When DocServer detects a meaningful change to an OAS, a new iteration is generated and stored as an artifact in the database. These artifacts allow for easy comparison between different iterations and also enable OAS changes to be displayed in a human readable format.
+
+For the local DEV environment these artifacts can be viewed with the following URL pattern:
 
 ```http://<LOCALSTACK_HOST>/<S3_BUCKET>/oasdiff/<API_NAME>_v<API_VERSION>_i<COMPARISON_ITERATION#>_i<ITERATION#>.html?minIter=<OLDEST_ITERATION_STORED>&maxIter=<NEWEST_ITERATION_STORED>```
```
Comment (mheusgit) [inline README.md:338]: With your change these instructions should no longer apply

File: README.md [hunk 2]
```diff
@@ -261,54 +285,65 @@ A sample kong service and OAS are provided for local end-to-end testing. The `do
...
 
 Example: `http://localhost:4566/oas-bucket/oasdiff/one_ring_v1_i0_i1.html?minIter=1&maxIter=2`
 
 The `localstack` S3 CSS & JS file is required for this to function correctly.  Running the following will get these in place for testing.
+
 ```
-aws --endpoint-url=http://localhost:4566 s3 cp ./dev/oasdiff/oasdiff.css s3://oas-bucket/oasdiff/
-aws --endpoint-url=http://localhost:4566 s3 cp ./dev/oasdiff/oasdiff.js s3://oas-bucket/oasdiff/
+aws --endpoint-url=http://localhost:4566 s3 cp ./src/ui/oasdiff/oasdiff.css s3://oas-bucket/oasdiff/
```
Comment (mheusgit) [inline README.md:345]: With your change these S3 population steps should no longer apply

File: src/app.ts [hunk 3]
```diff
@@ -46,6 +47,17 @@ export default async function configureApp(
 
   const app = express();
 
+  // Upload oasdiff.js file to s3
+  if (!cfg.IS_DEV) {
```
[the remark this answers is not in the document]
Comment (mheusgit) [reply src/app.ts:51]: Maybe it should be based on "if (cfg.IS_DEV_NODE)"?

File: src/utils/s3.service.ts [hunk 4]
```diff
@@ -130,6 +131,48 @@ class S3Service {
     }
   };
 
+  uploadOasDiff = async (
```
Comment (mheusgit) [inline src/utils/s3.service.ts:134]: Move method and logic connected to content setup to portalStore.service.ts.  
Call the "await this.s3Service.upload(key, content, fileType);" in method to use S3

File: jest.config.js [hunk 5]
```diff
@@ -18,7 +18,7 @@ module.exports = {
     '.model.ts',
     '/dist-test/',
   ],
-  testTimeout: 30000,
+  testTimeout: 50000,
```
Comment (mheusgit) [inline jest.config.js:21]: Team agreed on this temp fix.  Long term fix outside of ticket scope.

File: src/portalStore/portalStore.service.ts [hunk 6]
```diff
@@ -42,6 +43,20 @@ class PortalStoreService {
     await this.s3Service.upload(key, content, 'text/html');
   }
 
+  public async uploadOASDiffFiles(): Promise<void> {
```
Comment (mheusgit) [inline src/portalStore/portalStore.service.ts:46]: Nice reusable code

**Comment by mheusgit** [comment]
Fired off another build.  Last one still ran into the jest timeout.

**Comment by mheusgit** [comment]
For some reason when I pulled this down locally to verify it didn't work.

**Comment by mheusgit** [comment]
The file data is saved on the wrong format.

**Comment by mheusgit** [comment]
Assets are working as needed now

**Review by dsteeve — COMMENTED** [review]
just a small change request to the readme.  looks good!

File: README.md [hunk 7]
```diff
@@ -261,56 +285,60 @@ A sample kong service and OAS are provided for local end-to-end testing. The `do
...
-For local DEV environment these static files can be viewed with the following URL pattern: 
+When DocServer detects a meaningful change to an OAS, a new iteration is generated and stored as an artifact in the database. These artifacts allow for easy comparison between different iterations and also enable OAS changes to be displayed in a human readable format.
+
+For the local DEV environment these artifacts can be viewed with the following URL pattern:
 
 ```http://<LOCALSTACK_HOST>/<S3_BUCKET>/oasdiff/<API_NAME>_v<API_VERSION>_i<COMPARISON_ITERATION#>_i<ITERATION#>.html?minIter=<OLDEST_ITERATION_STORED>&maxIter=<NEWEST_ITERATION_STORED>```
 
 Example: `http://localhost:4566/oas-bucket/oasdiff/one_ring_v1_i0_i1.html?minIter=1&maxIter=2`
 
-The `localstack` S3 CSS & JS file is required for this to function correctly.  Running the following will get these in place for testing.
```
[the remark this answers is not in the document]
Comment (jeremyhwu) [reply README.md:308]: Added this line back in

**Review by mheusgit — APPROVED** [review]
Looks good

**Pull request merged**